src/test/java/dev/openfeature/sdk/EventProviderTest.java

```diff
@@ -75,6 +80,110 @@ void doesNotThrowWhenOnEmitSame() {
         eventProvider.attach(onEmit2); // should not throw, same instance. noop
     }
 
+    @Test
+    @SneakyThrows
+    @Timeout(value = 2, threadMode = Timeout.ThreadMode.SEPARATE_THREAD)
+    @DisplayName("should not deadlock on emit called during emit")
+    void doesNotDeadlockOnEmitStackedCalls() {
+        StackedEmitCallsProvider provider = new StackedEmitCallsProvider();
+        OpenFeatureAPI.getInstance().setProviderAndWait(provider);
+    }
+
+    static class StackedEmitCallsProvider extends EventProvider {
```

Comment:
@toddbaert @aepfli I'm not super happy with the location of the test and related code as it seems to bload this test class quite a bit, my intuition would be to move it out, but I don't see a good location, I was considering to move the support code to `fixtures` but not sure, any suggestion?

Reply:
There's a "testutils" folder which has some utilities as well as some implementations (TestEventsProvider.java); you could put it there.